engine/cmd/pull.go

```diff
@@ -127,26 +119,23 @@ func runServer(ctx context.Context, c *config.Config, socketPath string) error {
 		}
 	})
 
-	srv := http.Server{Handler: mux}
-	l, err := net.Listen("unix", socketPath)
-	if err != nil {
-		return fmt.Errorf("listen: %s", err)
+	srv := http.Server{
+		Addr:    fmt.Sprintf(":%d", port),
+		Handler: mux,
 	}
-	defer func() {
-		if err := l.Close(); err != nil {
-			log.Printf("close: %v\n", err)
-		}
-	}()
-
 	ctx, cancel := context.WithCancel(ctx)
 	go func() {
 		defer cancel()
-		log.Printf("Starting HTTP server at %q\n", socketPath)
-		if err := srv.Serve(l); err != http.ErrServerClosed {
+		log.Printf("Starting HTTP server at %q\n", srv.Addr)
+		if err := srv.ListenAndServe(); err != http.ErrServerClosed {
 			log.Printf("serve: %v\n", err)
 		}
 	}()
 
+	if modelID != "" {
```

Comment:
Just wanted to check my understanding. When the deamon mode is enabled, specifying a model ID is optional. When a model ID is specified, the model is pulled at the deamon startup time. 

Maybe nice to have some comment somewhere.

Reply:
Yes, that's correct. I'll add a comment here.